app/Http/Controllers/JobController.php

```diff
@@ -150,4 +151,33 @@ public function acceptOrReject($userId, $jobId, $status)
             'users' => $users,
         ]);
     }
+
+    public function filter(Request $request)
+    {
+        if (is_null($request->tag)){
+            $jobs = Job::all();
+            foreach ($jobs as $job) {
+                $jobs->this = Job::findOrFail($job->id);
+            }
+
+            return view('layouts.all_job', [
+                'jobs' => $jobs,
+            ]);
+        }
+        $suitableJobs = DB::table('jobs')
+            ->join('taggables', 'jobs.id', '=', 'taggables.taggable_id')
+            ->join('tags', 'tags.id', '=', 'taggables.tag_id')
+            ->select('jobs.id')->whereIn('tags.id',$request->tag)
+            ->where('taggable_type','App\\Models\\Job')
+            ->groupBy('jobs.id')
+            ->havingRaw('count(jobs.id)='.count($request->tag))
+            ->get();
+        foreach ($suitableJobs as $suitableJob) {
+            $suitableJob->this = Job::findOrFail($suitableJob->id);
+        }
```

Comment:
`$suitableJob->this` là gì nhỉ?

Reply:
Em lấy các công việc có các Tag mình chọn bằng truy vấn groupBy('jobs.id'), having count('jobs.id') như ý tưởng hôm demo lần 1 chị nói, truy vấn này chỉ lấy ra được thuộc tính Id của các Job cần tìm chứ không lấy ra đc các model Job cần tìm, em chưa nghĩ ra cách lấy model Job lên phải foreach rồi truyền model Job vào this của $suitableJob (đã sửa thành $filterJob). Tại view khi truy vấn $suitableJob->this sẽ lấy ra được model Job cần tìm

Reply:
Đã xử lý đc